code/modules/library/book.dm

```diff
@@ -103,6 +103,15 @@
 /obj/item/book/proc/on_read(mob/user)
 	if(book_data?.content)
 		user << browse("<meta charset=UTF-8><TT><I>Penned by [book_data.author].</I></TT> <BR>" + "[book_data.content]", "window=book[window_size != null ? ";size=[window_size]" : ""]")
+
+		if(ishuman(user))
+			var/mob/living/carbon/human/reader = user
+			LAZYINITLIST(reader.mind.book_titles_read)
+			var/has_not_read_book = isnull(reader.mind.book_titles_read[starting_title])
+
+			if(has_not_read_book) // any new books give bonus mood
+				SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "book_nerd", /datum/mood_event/book_nerd)
+			reader.mind.book_titles_read[starting_title] = TRUE
```

Comment:
Since you moved it from human to mind, you can get rid of the human typecheck, and just check `user.mind` instead.

Reply:
Does the `COMSIG_ADD_MOOD_EVENT` not care if the user is not human?

Reply:
`COMSIG_ADD_MOOD_EVENT` is a signal, so it doesn't remotely care what it's being sent to. 
If the atom you're sending it to has no mood component, nothing happens.

Reply:
Refactored with https://github.com/tgstation/tgstation/pull/66371/commits/c5eac09bf059c035ae8ec37309547f62b0638f80